domain/src/main/java/life/qbic/domain/usermanagement/UserDomainService.java

```diff
@@ -34,15 +36,51 @@ public UserDomainService(UserRepository userRepository) {
    */
   public void createUser(String fullName, String email, char[] rawPassword) throws UserException {
     // First check, if a user with the provided email already exists
-    if (userRepository.findByEmail(email).isPresent()) {
-      throw new UserException("User with email address already exists.");
-    }
+    validateUserCredentials(fullName, email, rawPassword);
     var domainEventPublisher = DomainEventPublisher.instance();
     var user = User.create(fullName, email);
     user.setPassword(rawPassword);
     userRepository.addUser(user);
-
     var userCreatedEvent = UserRegistered.create(user.getId(), user.getFullName(), user.getEmail());
     domainEventPublisher.publish(userCreatedEvent);
   }
+
+  private boolean isEmailInDatabase(String email) {
+    return userRepository.findByEmail(email).isPresent();
+  }
+
+  private void validateUserCredentials(String fullName, String email, char[] rawPassword) {
+    List<UserException> userExceptions = new ArrayList<>();
+    var user = new User();
+    if (isEmailInDatabase(email)) {
+      userExceptions.add(new UserException("User with email address already exists."));
+    }
+    try {
+      user.setFullName(fullName);
+    } catch (UserException e) {
+      userExceptions.add(e);
+    }
+    try {
+      user.setEmail(email);
+    } catch (UserException e) {
+      userExceptions.add(e);
+    }
+    try {
+      user.setPassword(rawPassword);
+    } catch (UserException e) {
+      userExceptions.add(e);
+    }
+    throwCorrectUserException(userExceptions);
+  }
```

Comment:
This does not belong here. The user itself has the responsibility to check it's parameters upon creation. That is why we have a static method in the user for creating users. We do not intent the constructor of a User to be used outside of the User class.